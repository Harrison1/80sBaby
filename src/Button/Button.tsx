import * as React from 'react'
import styled, { keyframes } from '../utils/styled-components'

interface ButtonProps {
  autoFocus?: boolean
  ariaLabel?: string
  children?: React.ReactNode
  disabled?: boolean
  fontSize?: string
  name?: string
  primary?: boolean
  onClick?: () => void
  type?: string
  value?: string
}

const gradient = keyframes`
  0% {
    background-position: 0% 50%
  }
  50% {
    background-position: 100% 50%
  }
  100% {
    background-position: 0% 50%
  }
`

const SButton = styled(`button`)<ButtonProps>`
  background: linear-gradient(-45deg, #ee7752, #e73c7e, #23a6d5, #23d5ab);
  background-size: 400% 400%;
  border: none;
  border-radius: 999rem;
  cursor: pointer;
  font-size: ${props => (props.fontSize ? props.fontSize : '1rem')};
  padding: 4px;
  animation: ${gradient} 15s ease infinite;
  animation-play-state: paused;
  &:hover {
    animation-play-state: running;
    will-change: background;
    div {
      background: rgba(255, 255, 255, 0.05);
      color: #fff;
    }
  }
  &:active,
  &:focus,
  &:visited {
    border-color: #80bdff;
    outline: 0;
    box-shadow: 0 0 0 0.3rem rgba(180, 0, 255, 0.35);
  }
  div {
    background: #fff;
    border-radius: 999rem;
    padding: 10px 20px;
    transition: background 0.5s ease, color 0.5s ease;
  }
`

const noop = () => {}

const Button = (props: ButtonProps) => (
  <SButton
    ariaLabel={props.ariaLabel ? props.ariaLabel : props.name}
    autoFocus={props.autoFocus}
    disabled={props.disabled}
    name={props.name}
    onClick={!props.disabled ? props.onClick : noop}
    type={props.type}
    value={props.value}
    primary={props.primary ? true : false}
    fontSize={props.fontSize}
  >
    <div>{props.children}</div>
  </SButton>
)

export default Button
