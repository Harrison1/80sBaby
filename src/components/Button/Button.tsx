import * as React from 'react';
import styled, { keyframes } from '../../utils/styled-components'

interface ButtonProps {
  autoFocus?: boolean;
  ariaLabel?: string;
  children?: React.ReactNode;
  disabled?: boolean;
  fontSize?: string;
  name?: string;
  primary?: boolean;
  onClick?: () => void;
  type?: string;
  value?: string;
}

const gradient = keyframes`
  0% {
    background-position: 0% 50%
  }
  50% {
    background-position: 100% 50%
  }
  100% {
    background-position: 0% 50%
  }
`

const SButton = styled(`button`)<ButtonProps>`
  background: linear-gradient(-45deg, #EE7752, #E73C7E, #23A6D5, #23D5AB);
  background-size: 400% 400%;
  border: none;
  border-radius: 3rem;
  cursor: pointer;
  font-size: ${props => props.fontSize ? props.fontSize : '1rem'};
  padding: 4px;
  &:hover {
    animation: ${gradient} 15s ease infinite;
    will-change: background;
  }
  &:active, 
  &:focus, 
  &:visited {
    border-color: #80bdff;
    outline: 0;
    box-shadow: 0 0 0 0.3rem rgba(180, 0, 255, 0.35);
  }
  div {
    background: #fff;
    border-radius: 3rem;
    padding: 10px 20px;
  }
`
const noop = () => {}

const Button = (props: ButtonProps) => 
  <SButton
    aria-label={props.ariaLabel}
    autoFocus={props.autoFocus}
    disabled={props.disabled}
    name={props.name}
    onClick={!props.disabled ? props.onClick : noop}
    type={props.type}
    value={props.value}
    primary={props.primary ? true : false}
    fontSize='10px'
  >
    <div>
      { props.children }
    </div>
  </SButton>

export default Button
