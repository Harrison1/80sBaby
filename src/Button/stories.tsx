import * as React from 'react'
import { withKnobs, text, boolean } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import Button from './Button'

const stories = storiesOf('Button', module)

stories.addDecorator(withKnobs)

stories
  .add('with text', () => (
    <Button
      aria-label={text('Aria Label', 'My Button')}
      autoFocus={boolean('Auto Focus', false)}
      disabled={boolean('Disabled', false)}
      fontSize={text('Font Size', '1rem')}
      name={text('Name', 'Galaxy')}
      type={text('Type', 'button')}
      value={text('Value', '1')}
    >
      {text('Button Text', 'Hello Galaxy')}
    </Button>
  ))
