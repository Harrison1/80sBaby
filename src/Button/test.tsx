import * as React from 'react'
import Button from './Button'
import * as renderer from 'react-test-renderer'

test('Button changes the class when hovered', () => {
  const component = renderer.create(<Button>80s Baby</Button>)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
