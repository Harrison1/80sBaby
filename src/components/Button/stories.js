import React from 'react';
import { storiesOf } from '@storybook/react';
import Button from './Button';

storiesOf('Button')
	.add('with text', () => (
		<Button>Hello Galaxy</Button>
	))
	.add('with emoji', () => (
		<Button>🚿 🚿 🐈 🐈</Button>
  ))
  