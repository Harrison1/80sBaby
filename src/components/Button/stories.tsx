import * as React from 'react';
import { storiesOf } from '@storybook/react';
import Button from './Button';

const stories = storiesOf('Button', module);

stories
	.add('with text', () => (
		<Button>Hello Galaxy</Button>
	))
	.add('with emoji', () => (
		<Button>🚿 🚿 🐈 🐈</Button>
  ))
  