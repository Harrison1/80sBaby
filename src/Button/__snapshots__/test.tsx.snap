// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button changes the class when hovered 1`] = `
<button
  className="sc-bdVaJa gWCEdI"
>
  <div>
    80s Baby
  </div>
</button>
`;
